kv/dist_sender.go

```diff
@@ -304,20 +304,17 @@ func (ds *DistSender) getNodeDescriptor() *roachpb.NodeDescriptor {
 func (ds *DistSender) sendRPC(trace *tracer.Trace, rangeID roachpb.RangeID, replicas replicaSlice, order rpc.OrderingPolicy,
 	ba roachpb.BatchRequest) (*roachpb.BatchResponse, error) {
 	if len(replicas) == 0 {
-		return nil, util.Errorf("replicas set is empty")
+		return nil, noNodeAddrsAvailError{}
 	}
 
 	// Build a slice of replica addresses (if gossiped).
```

Comment:
This comment has slightly rotted (I think some earlier version of the code constructed a `replicaSlice` here, and between above and below it looked up addresses from Gossip, discarding nodes which weren't addressable).